src/main/jit/CompiledExpression.cpp

```diff
@@ -86,7 +87,10 @@ CompiledExpression::CompiledExpression(const Closure* closure)
     // function->dump(); // So we can see what's going on while developing.
     llvm::verifyFunction(*function);
 
-    // TODO: add optimization passes and re-verify.
+    // Run some optimization passes.
+    RemoveRedundantCallsToSetVisibility(module.get(), function);
+    // TODO(ArunChauhan): Uncomment the following when it's more effective.
+    // BasicIntraProceduralOptimizations(module.get(), function);
```

Comment:
Ok, but it really does nothing on the IR coming out of the R AST-to-LLVM compiler.  OTOH, it might help if we start JIT-compiling the IR coming from C/C++.